### PR TITLE
Pass through unsupported user data during PDU parsing. Connected to #242

### LINKED ISSUE #242: Pass through unsupported user data during PDU parsing.
For example, AAssociateRQ and AAssociateAC currently just skip any unsupported User Data, leaving it inaccessible to implementers of IDicomServiceProvider.  Instead, it would be nice to either pass the raw data through in the DicomAssociation, or to provide a way to register custom parsers for unsupported/unimplemented data, so that developers can get access to it without needing to modify core code.

### PR DESCRIPTION
**Pull request by deyung**
Fixes #242 

Changes proposed in this pull request:
- Added delegate to pass bytes back to the implementer.
-
-

### CODE REVIEW

File: DICOM/Network/DicomAssociation.cs [hunk 1]
```diff
@@ -91,6 +91,11 @@ public DicomAssociation(string callingAe, string calledAe, uint maxPduLength = 1
         /// </summary>
         public List<DicomExtendedNegotiation> ExtendedNegotiations { get; private set; }
 
+		/// <summary>
+		/// Callback to handle unsupported user information
```
Comment (anders9ustafsson) [inline DICOM/Network/DicomAssociation.cs]: @deyung Please provide a complete API documentation for the delegate, including argument documentation, and a separate documentation section for the event. Please also use the *Delegate* postfix on the delegate, e.g. `HandlePDUBytesDelegate`.

**Comment by zcr01** [comment]
@anders9ustafsson @deyung  I think it's on the right path re: #242 but unless I'm missing something, I'm not sure how the IDicomServiceProvider would subscribe to the event until it is too late and processing is already complete.

**Comment by deyung** [comment]
@zcr01 it could be used in the override of OnReceiveAssociationRequest(DicomAssociation association) implementation. Would that be too late?

**Comment by zcr01** [comment]
See here: https://github.com/fo-dicom/fo-dicom/blob/f5eb945044ae679c3715be8317340ca95a07d7af/DICOM/Network/DicomService.cs#L355-L371

A new PDU object is created, Read() is called on it, and after that's done it's passed on up through OnReceiveAssociationRequest().

**Comment by deyung** [comment]
Alright, I'll make some changes.

**Pull request merged**